contracts/core/BorrowableMarginPool.sol

```diff
@@ -0,0 +1,288 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity =0.6.10;
+
+import {ERC20Interface} from "../interfaces/ERC20Interface.sol";
+import {OtokenInterface} from "../interfaces/OtokenInterface.sol";
+import {AddressBookInterface} from "../interfaces/AddressBookInterface.sol";
+import {WhitelistInterface} from "../interfaces/WhitelistInterface.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+import {SafeERC20} from "../packages/oz/SafeERC20.sol";
+import {MarginPool} from "./MarginPool.sol";
+
+/**
+ * @author Ribbon Team
+ * @title MarginPoolV2
+ * @notice Contract that holds all protocol funds AND allows collateral borrows
+ */
+contract BorrowableMarginPool is MarginPool {
+    using SafeMath for uint256;
+    using SafeERC20 for ERC20Interface;
+
+    uint16 public constant TOTAL_PCT = 10000; // Equals 100%
+
+    /// @dev mapping between collateral asset and borrow PCT
+    mapping(address => uint256) public borrowPCT;
+    /// @dev mapping between address and whitelist status of borrower
+    mapping(address => bool) internal whitelistedBorrower;
+    /// @dev mapping between address and whitelist status of otoken buyer
+    /// This is the whitelist for all the holders of oTokens that have a claim to
+    /// collateral in this pool
+    mapping(address => bool) internal whitelistedOTokenBuyer;
+    /// @dev mapping between address and whitelist status of options vault
+    /// This denotes whether an options vault can create a margin vault with
+    /// collateral custodied in this borrowable vault
+    mapping(address => bool) internal whitelistedOptionsVault;
```

Comment:
Sorry if it seems there are alot of naming comments, but let's try to use `borrowableVault` and `borrowablePool` terminology everywhere, so make sense here IMO to rename it to `whitelistedBorrowableVault`, as also vault type 2 indicate a borrowable vault.

Reply:
This was a specific decision since you are not exactly borrowing from the options vault, you are borrowing from the margin pool that the options vault deposits into. So "whitelistedBorrowableVault" when describing a DOV feels like a misnomer at worst and confusing at best.